local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs

```diff
@@ -0,0 +1,453 @@
+        .get();
+
+    test_ctx
+        .build_tx_instantiate2()
+        .with_label(&format!("valence_osmo_liquid_pooler_{label_seed}"))
+        .with_code_id(pooler_code_id)
+        .with_salt_hex_encoded(&hex::encode(salt))
+        .with_msg(
+            serde_json::to_value(&valence_osmo_liquid_pooler::msg::InstantiateMsg {
+                op_mode_cfg: ContractOperationModeConfig::Permissionless,
+                holder_address: holder.to_owned(),
+                note_address: polytone_note_address,
+                pool_id: Uint64::default(),
+                osmo_ibc_timeout: Uint64::new(DEFAULT_TIMEOUT),
+                party_1_chain_info: PartyChainInfo {
+                    party_chain_to_neutron_channel: osmosis_to_neutron.clone(),
+                    neutron_to_party_chain_channel: neutron_to_osmosis.clone(),
+                    outwards_pfm: None,
+                    inwards_pfm: None,
+                    ibc_timeout: Uint64::new(DEFAULT_TIMEOUT),
+                },
+                party_2_chain_info: PartyChainInfo {
+                    party_chain_to_neutron_channel: osmosis_to_neutron,
+                    neutron_to_party_chain_channel: neutron_to_osmosis.clone(),
+                    outwards_pfm: None,
+                    inwards_pfm: None,
+                    ibc_timeout: Uint64::new(DEFAULT_TIMEOUT),
+                },
+                osmo_to_neutron_channel_id: neutron_to_osmosis,
+                party_1_denom_info: PartyDenomInfo {
+                    osmosis_coin: Coin::new(0, "uosmo"),
+                    local_denom: "untrn".to_owned(),
+                },
+                party_2_denom_info: PartyDenomInfo {
+                    osmosis_coin: Coin::new(0, "uosmo"),
+                    local_denom: "untrn".to_owned(),
+                },
+                osmo_outpost: osmo_outpost_addr,
+                lp_token_denom: "uxyzabcd".into(),
+                funding_duration: CwDuration::Time(0),
+                single_side_lp_limits: SingleSideLpLimits {
+                    asset_a_limit: Default::default(),
+                    asset_b_limit: Default::default(),
+                },
+                slippage_tolerance: None,
+                pool_price_config: PoolPriceConfig {
+                    expected_spot_price: Decimal::one(),
+                    acceptable_price_spread: Decimal::from_ratio(Uint128::one(), Uint128::new(2)),
+                },
+            })
+            .unwrap(),
+        )
+        .with_flags("--gas 1000000")
+        .send()
+        .unwrap();
+
+    Ok(pooler_address)
+}
+
+/// Tests that the pooler does not advance when there is no relayer available.
+/// Leaves the relayer intact after its execution.
+fn test_pooler_timeout(test_ctx: &mut TestContext, pooler: &str) -> Result<(), LocalError> {
+    // Stop the relayer
+    stop_relayer(test_ctx);
+
+    let neutron = test_ctx.get_chain(NEUTRON_CHAIN_NAME);
+
+    let pooler_contract =
```

Comment:
since you already have the contract address (you are passing it as an argument). You can reuse my helper function I created in common.rs to query the contract state, up to you, but you can remove a lot of code here that way. I also made one for ticking so you can remove pretty much all the tick/query code here and use those.
I forgot to mention this in other PRs because you also query state / tick there.
This will also remove a lot of duplication everywhere.

Reply:
Pushed the change! Going to fix this redundancy in the other tests too in a PR after this one.

Reply:
Just ran the tests locally btw on the latest commit. Everything seems to be working!

Reply:
nice!